client/chat/channel.tsx

```diff
@@ -171,151 +167,121 @@ const UserListEntry = React.memo<UserListEntryProps>(props => {
   )
 })
 
-const UsersVirtualizedList = styled(VirtualizedList)`
-  &:focus,
-  & > div:focus {
-    outline: none;
-  }
-`
-
 interface UserListProps {
   users: UsersRecord
 }
 
-class UserList extends React.Component<UserListProps> {
-  state = {
-    width: 0,
-    height: 0,
-  }
-  _contentRef = React.createRef<HTMLDivElement>()
-  _listRef = React.createRef<VirtualizedList>()
-
-  componentDidMount() {
-    this.updateDimensions()
-  }
-
-  componentDidUpdate(prevProps: UserListProps) {
-    if (prevProps.users !== this.props.users) {
-      this._listRef.current?.recomputeRowHeights()
-    }
-  }
+const UserList = React.memo((props: UserListProps) => {
+  const { active, idle, offline } = props.users
+  const [dimensionsRef, containerRect] = useObservedDimensions()
+  const listRef = useRef<VariableSizeList | null>(null)
+  const containerCallback = useCallback(
```

Comment:
What I meant here is that you don't actually need this callback at all, you can simply pass `dimensionsRef` as the `ref` on the component. When we're creating a function like this in other places, it's so that we can store a `ref` value in 2 or more places.